lib/evaluation/handlers/limitDisclosureEvaluationHandler.ts

```diff
@@ -48,15 +48,26 @@ export class LimitDisclosureEvaluationHandler extends AbstractEvaluationHandler
     const limitDisclosure = constraints.limit_disclosure as Optionality;
     verifiableCredential.forEach((vc, index) => {
       if (this.limitDisclosureSupported(vc, index, idIdx)) {
-        const verifiableCredentialToSend = this.createVcWithRequiredFields(vc, fields, idIdx, index);
-        if (verifiableCredentialToSend) {
-          verifiableCredential[index] = verifiableCredentialToSend;
-          this.createSuccessResult(idIdx, `$[${index}]`, limitDisclosure);
-        }
+        this.limitDisclosureEnforcement(vc, fields, idIdx, index, verifiableCredential, limitDisclosure);
       }
     });
   }
 
+  private limitDisclosureEnforcement(
+    vc: VerifiableCredential,
+    fields: Field[],
+    idIdx: number,
+    index: number,
+    verifiableCredential: VerifiableCredential[],
+    limitDisclosure: 'required' | 'preferred'
+  ) {
+    const verifiableCredentialToSend = this.createVcWithRequiredFields(vc, fields, idIdx, index);
+    if (verifiableCredentialToSend) {
+      verifiableCredential[index] = verifiableCredentialToSend;
+      this.createSuccessResult(idIdx, `$[${index}]`, limitDisclosure);
+    }
```

Comment:
What if !verifialbeCredentialToSend?

Reply:
The credential will remain untouched and an error will be raised. The vc won't be submitted.

Reply:
Please make a remark/comment about it. Because the code is scattered across, developers not into the code would have similar questions

Reply:
Fixed!